lib/linter.js

```diff
@@ -74,12 +74,19 @@ module.exports = function (cwd, args, text) {
 
     return JSON.stringify(fileConfig, null, '  ');
   }
+  if (currentOptions.fixToStdout && !stdin) {
+    return 'The --fix-to-stdout option must be used with --stdin.'
+      + '\n# exit 1';
+  }
   var report;
   if (stdin && text) {
     report = engine.executeOnText(text, currentOptions.stdinFilename);
   } else {
     report = engine.executeOnFiles(files);
   }
+  if (currentOptions.fixToStdout) {
+    return report.results[0].output || text;
```

Comment:
Globs are expanded by eslint. Let's take the easy path then and only support `--stdin` then. If text is empty, we just do nothing.